Userland/Libraries/LibJS/Runtime/ArrayConstructor.cpp

```diff
@@ -151,7 +149,7 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayConstructor::from)
 
             Value mapped_value;
             if (map_fn) {
-                mapped_value = vm.call(*map_fn, this_arg, next_value, Value(k));
+                mapped_value = TRY_OR_DISCARD(vm.call(*map_fn, this_arg, next_value, Value(k)));
                 if (vm.exception()) {
                     iterator_close(*iterator);
                     return {};
```

Comment:
Behavior here is slightly more complicated on exception so this cannot just be `TRY_OR_DISCARD`